### PR TITLE
[autoparallel] support distributed dataloader option

### PR DESCRIPTION
**Pull request by YuliangLiu0306**

### CODE REVIEW

File: colossalai/auto_parallel/tensor_shard/node_handler/strategy/output_generator.py [hunk 1]
```diff
@@ -48,5 +64,47 @@ def collate_strategies(self) -> List[ShardingStrategy]:
         strategy = self.get_sharding_strategy(name=name,
                                               sharding_spec_mapping=sharding_spec_mapping,
                                               communication_action_mapping=communication_action_mapping)
+        return strategy
+
+    def distributed_strategy(self, mesh_list: List[List[int]] = None):
```
Comment (feifeibear) [inline colossalai/auto_parallel/tensor_shard/node_handler/strategy/output_generator.py:69]: return type hint
-> List[XXX]

File: colossalai/auto_parallel/tensor_shard/node_handler/strategy/output_generator.py [hunk 2]
```diff
@@ -32,7 +45,10 @@ def update_memory_cost(self, strategy: ShardingStrategy):
         memory_cost = TrainCycleItem(fwd=fwd_mem_cost, bwd=bwd_mem_cost, total=total_mem_cost)
         strategy.memory_cost = memory_cost
 
-    def collate_strategies(self) -> List[ShardingStrategy]:
+    def replica_strategy(self):
```
Comment (feifeibear) [inline colossalai/auto_parallel/tensor_shard/node_handler/strategy/output_generator.py:48]: return type hint

**Review by feifeibear — APPROVED** [review]

File: colossalai/auto_parallel/tensor_shard/node_handler/strategy/output_generator.py (hunk 1 above)
Comment (YuliangLiu0306) [reply colossalai/auto_parallel/tensor_shard/node_handler/strategy/output_generator.py]: Done

File: colossalai/auto_parallel/tensor_shard/node_handler/strategy/output_generator.py (hunk 2 above)
Comment (YuliangLiu0306) [reply colossalai/auto_parallel/tensor_shard/node_handler/strategy/output_generator.py]: Done

**Pull request merged**